packages/core/frontend/sirius-components-core/src/onboarding/NewDocumentArea.tsx

```diff
@@ -0,0 +1,206 @@
+ *******************************************************************************/
+import { gql, useMutation } from '@apollo/client';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import IconButton from '@material-ui/core/IconButton';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Snackbar from '@material-ui/core/Snackbar';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import { Close as CloseIcon, NoteAdd } from '@material-ui/icons';
+import { useEffect, useState } from 'react';
+import { v4 as uuid } from 'uuid';
+import {
+  GQLCreateDocumentMutationInput,
+  GQLCreateDocumentMutationVariables,
+  GQLInvokeEditingContextActionInput,
+  GQLInvokeEditingContextActionVariables,
+  NewDocumentAreaProps,
+} from './NewDocumentArea.types';
+
+const useNewDocumentAreaStyles = makeStyles(() => ({
+  card: {
+    backgroundColor: 'transparent',
+
+    maxWidth: '330px',
+    minWidth: '330px',
+  },
+  cardContent: {
+    height: '100%',
+    maxHeight: '400px',
+    overflowY: 'auto',
+  },
+}));
+
+const createDocumentMutation = gql`
+  mutation createDocument($input: CreateDocumentInput!) {
+    createDocument(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const invokeEditingContextActionMutation = gql`
+  mutation invokeEditingContextAction($input: InvokeEditingContextActionInput!) {
+    invokeEditingContextAction(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+export const NewDocumentArea = ({
+  editingContextId,
+  stereotypeDescriptions,
+  editingContextActions,
+  setSelection,
+  readOnly,
+}: NewDocumentAreaProps) => {
+  const classes = useNewDocumentAreaStyles();
+  const [message, setMessage] = useState('');
+  const [open, setOpen] = useState(false);
+
+  // Document creation
+  const [createDocument, { loading, data, error }] = useMutation(createDocumentMutation);
+  const onCreateDocument = (stereotypeDescriptionId: string) => {
+    const selected = stereotypeDescriptions.find((candidate) => candidate.id === stereotypeDescriptionId);
+    const input: GQLCreateDocumentMutationInput = {
+      id: uuid(),
+      editingContextId,
+      name: selected.documentName,
+      stereotypeDescriptionId,
+    };
+    const variables: GQLCreateDocumentMutationVariables = {
+      input,
+    };
+    createDocument({ variables });
+  };
+  useEffect(() => {
+    if (!loading && !error && data?.createDocument) {
+      const { createDocument } = data;
+      if (createDocument.__typename === 'ErrorPayload') {
+        setOpen(true);
+        setMessage(createDocument.message);
+      }
+    }
+  }, [loading, data, error, setSelection]);
+
```

Comment:
- Please consider potential errors using the variables `error`
- Why are you depending on setSelection here?